org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/DocumentProviderTest.xtend

```diff
@@ -136,24 +169,36 @@ class DocumentProviderTest extends AbstractGitTest {
 		remoteGit.assertSingleCommit(numberOfCommitsBefore, DELETE, existingFileName)
 	}
 
+	@Test
+	def void loadPullsChanges() {
+		// given
+		val expectedFileContents = "The file contents.\n"
+		val existingFileName = createAndPushFileToRepo("preExistingFile.txt", expectedFileContents)
+
+		// when
+		val actualFileContents = documentProvider.load(existingFileName)
+
+		// then
+		localGitRoot.root.assertFileExists(existingFileName)
+		actualFileContents.assertEquals(expectedFileContents)
+	}
+
+
+
 	private def createAndPushFileToRepo() {
-		val filename = "preExistingFile.txt"
-		remoteGitFolder.newFile(filename).createNewFile
-		remoteGit.add.addFilepattern(filename).call
-		remoteGit.commit.setMessage("set test preconditions").call
-		return filename
+		return this.createAndPushFileToRepo("preExistingFile.txt")
 	}
 
-	private def assertSingleCommit(Git git, int numberOfCommitsBefore, ChangeType expectedChangeType, String path) {
-		val numberOfCommitsAfter = git.log.call.size
-		numberOfCommitsAfter.assertEquals(numberOfCommitsBefore + 1)
-		val diffEntries = git.getDiffEntries(git.lastCommit)
-		git.getDiffEntries(git.lastCommit).exists [
-			changeType === expectedChangeType && newPath == path
-		].assertTrue('''Expected the following change: «expectedChangeType» «path», but found: «diffEntries.head.changeType» «diffEntries.head.newPath»''')
+	private def createAndPushFileToRepo(String path) {
+		return this.createAndPushFileToRepo(path, "These are the file's contents!\n")
 	}
-	
-	private def assertFileExists(File parent, String path) {
-		new File(parent, path).exists.assertTrue
+
+	private def createAndPushFileToRepo(String path, String fileContents) {
```

Comment:
What this method really does is `createAndWriteFileAddAndCommit`.
I would prefer creating two methods in the super class here instead.
1. `write(String path, String fileContents)`
2. `addAndCommit(Git git, String path, String message)`
This way we can reuse those methods.

Reply:
OK, here's what I did: I renamed the original methods `createPreExistingFileInRemoteRepository`, in accordance with your previous comment. I moved them to the base class, and also made them `protected` instead of `private`. I did not completely remove the methods for convenience and readability of the tests, though if you rather have them removed completely, I'm cool with that, too.

I also extracted the two methods you suggested, though I think (1) should either be `writeToRemoteGitFolder(…)`, or `write(TemporaryFolder targetDir, String path, String fileContents)`. I decided for the latter, because it makes a nice extension method.